pkg/lib/cockpit-components-table.jsx

```diff
@@ -228,8 +228,10 @@ export const ListingTable = ({
                             }
                         }} />
                     }
-                    {row.columns.map((cell, cellIndex) => {
+                    {row.columns.map((cell, prevCellsCnt) => {
                         const { key, ...cellProps } = cell.props || {};
+                        const prevCells = row.columns.slice(0, prevCellsCnt);
+                        const cellIndex = prevCells.reduce((acc, curr) => acc + (curr.props && curr.props.colSpan ? curr.props.colSpan : 1), 0);
```

Comment:
`curr.props?.colSpan || 1` ?

Reply:
Oh: and I suppose that colSpan won't ever be 0, right?  That wouldn't make sense... Because if it was 0, we'd end up rewriting it to 1.

Reply:
We dont specify 0 colspans. It's  >=1. colSpan 0 would rather be non present column.

Reply:
For the future: That's what `??` is for. `curr.props?.colSpan ?? 1` would  correctly return "0" if `colSpan == 0`.